common/quotas/global/collection/collection.go

```diff
@@ -163,89 +333,146 @@ func (c *Collection) backgroundUpdateLoop() {
+					counts := v.Collect()
+					if counts.Idle > gcAfterIdle || c.shouldDeleteKey(c.keyMode(globalKey), true) {
+						c.logger.Debug(
+							"deleting local ratelimiter",
+							tag.Dynamic("key", globalKey),
+							tag.Dynamic("idle", counts.Idle > gcAfterIdle),
+						)
+						c.local.Delete(k)
+						return true // continue iterating, possibly delete others too
+					}
+
+					c.sendMetrics(globalKey, "local", counts)
+					return true
+				})
+			}()
+
+			capacity := c.global.Len()
 			capacity += capacity / 10 // size may grow while we range, try to avoid reallocating in that case
-			usage := make(map[string]calls, capacity)
-			fallbacks, globals := 0, 0
-			c.limits.Range(func(k string, v *internal.FallbackLimiter) bool {
-				allowed, rejected, usingFallback := v.Collect()
-				if usingFallback {
-					fallbacks++
-				} else {
+			usage := make(map[string]rpc.Calls, capacity)
+			startups, failings, globals := 0, 0, 0
+			c.global.Range(func(k string, v *internal.FallbackLimiter) bool {
+				globalKey := c.km.LocalToGlobal(k)
+				counts, startup, failing := v.Collect()
+				if counts.Idle > gcAfterIdle || c.shouldDeleteKey(c.keyMode(globalKey), false) {
+					c.logger.Debug(
+						"deleting global ratelimiter",
+						tag.Dynamic("key", globalKey),
+						tag.Dynamic("idle", counts.Idle > gcAfterIdle),
+					)
+					c.global.Delete(k)
+					return true // continue iterating, possibly delete others too
+				}
+
+				if startup {
+					startups++
+				}
+				if failing {
+					failings++
+				}
+				if !(startup || failing) {
 					globals++
 				}
-				usage[k] = calls{
-					allowed:  allowed,
-					rejected: rejected,
+				usage[globalKey] = rpc.Calls{
+					Allowed:  counts.Allowed,
+					Rejected: counts.Rejected,
 				}
+				c.sendMetrics(globalKey, "global", counts)
+
 				return true
 			})
+
 			// track how often we're using fallbacks vs non-fallbacks.
-			// this also tells us about how many keys are used per host.
-			// timer-sums unfortunately must be divided by the update frequency to get correct totals.
-			c.scope.RecordTimer(metrics.GlobalRatelimiterFallbackUsageTimer, time.Duration(fallbacks))
-			c.scope.RecordTimer(metrics.GlobalRatelimiterGlobalUsageTimer, time.Duration(globals))
+			// with per-host metrics this can tell us if a host is abnormal
+			// compared to the rest of the cluster, e.g. persistent failing values.
+			c.scope.RecordHistogramValue(metrics.GlobalRatelimiterStartupUsageHistogram, float64(startups))
+			c.scope.RecordHistogramValue(metrics.GlobalRatelimiterFailingUsageHistogram, float64(failings))
+			c.scope.RecordHistogramValue(metrics.GlobalRatelimiterGlobalUsageHistogram, float64(globals))
+
+			if len(usage) > 0 {
+				sw := c.scope.StartTimer(metrics.GlobalRatelimiterUpdateLatency)
+				c.doUpdate(now.Sub(lastGatherTime), usage)
+				sw.Stop()
+			}
 
-			c.doUpdate(now.Sub(lastGatherTime), usage)
+			<-localMetricsDone // should be much faster than doUpdate, unless it's no-opped
```

Comment:
I _could_ wait on ctx.Done too, but did not currently because the local-limiter-metrics goroutine is not tracked any other way, and we'd want to wait for it when shutting down.

that can be done, it's just a moderate step more complicated, and the only latency in that goroutine should be due to local metric emission clogging up, which seems unlikely to be an issue.  and/or an issue _everywhere_ and not just here, so it hardly matters if this is slow.